### ISSUE TITLE
Windows WSL: user config ignored when xonsh is the default shell and launched explicitly

### ISSUE THREAD

**Issue by eugenesvk** [xonsh/xonsh#4495]
If I set xonsh to be the default shell on Windows Subsystem for Linux (version 2) via `chsh -s /home/linuxbrew/.linuxbrew/bin/xonsh`, it stops reading my `~/.config/xonsh/rc.xsh` file on manual startup via `"C:\WINDOWS\System32\wsl.exe" /home/linuxbrew/.linuxbrew/bin/xonsh`

However, in these two ways of invoking xonsh everything is fine, xonsh is reading my user config:
  - `"C:\WINDOWS\System32\wsl.exe"` is fine, 
  - `"C:\WINDOWS\System32\wsl.exe" -e  /home/linuxbrew/.linuxbrew/bin/xonsh` (where `-e` is bypassing the default shell)

In all three cases the `$XONSH_INTERACTIVE` and `$XONSH_LOGIN` variables are set to `1`

## xonfig

<details>

```
$ xonfig
xonfig
+------------------+----------------------+
| xonsh            | 0.10.1               |
| Git SHA          | e9b12c8b             |
| Commit Date      | Jul 24 22:47:37 2021 |
| Python           | 3.9.7                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.19               |
| shell type       | prompt_toolkit       |
| history backend  | json                 |
| pygments         | 2.9.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | unknown              |
| on wsl           | True                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
| xontrib          | []                   |
+------------------+----------------------+
```

</details>

## Expected Behavior

Xonsh always reads the user config files unless a `--no-rc` flag is passed

## Current Behavior
In some cases, the user config files are ignored, see above/below for more details

### Traceback (if applicable)

<details>

```
traceback
```

</details>

## Steps to Reproduce
- (console window`1`) launch a WSL2 shell
- set any other shell to be the default, e.g. `chsh -s /usr/bin/bash`
- (window`2`) launch a new `cmd.exe` window`2`
- launch xonsh via `"C:\WINDOWS\System32\wsl.exe" /home/linuxbrew/.linuxbrew/bin/xonsh`
- observe that user config files are read
- (window`1`) set xonsh as the default shell via `chsh -s /home/linuxbrew/.linuxbrew/bin/xonsh`
- (windows`2`) exit xonsh and launch it again via `"C:\WINDOWS\System32\wsl.exe" /home/linuxbrew/.linuxbrew/bin/xonsh`
- observe that no user config is read, you get a fresh welcome message

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
## Why RC files are not loaded when xonsh is the default shell in WSL                                    
                                               
  ### What happens                                                                                         
   
  When `wsl.exe /path/to/xonsh` is called and xonsh is already the default shell (via `chsh`), WSL creates 
  a double invocation: the login shell spawns another xonsh instance. This second instance may receive a
  non-TTY stdin or be treated as a script, putting xonsh into **non-interactive mode**.                    
                                                                  
  In non-interactive mode, `~/.xonshrc` is intentionally excluded from loading. This is the same convention
   used by other shells: bash, zsh, fish.
                                                                                                                                                                                                                      
  ### RC files that always load
                                                                                                           
  Not all RC files are filtered. The default `$XONSHRC` includes: `/etc/xonsh/xonshrc`, `~/.config/xonsh/rc.xsh`.                                                
   
  **Solution:** move your configuration to `~/.config/xonsh/rc.xsh`. This file is loaded regardless of the 
  shell mode.                                                     
                                                                                                           
  ### A note on using xonsh as a login shell                      

  Xonsh is **not recommended as a login shell**. It is not POSIX-compliant, which means system scripts,    
  session managers, and tools like `ssh` that expect `/bin/sh`-compatible behavior may break. Many system
  startup scripts (e.g. `/etc/profile`, `~/.profile`) source POSIX shell syntax that xonsh cannot parse.   
                                                                  
  Instead, keep a POSIX shell (bash, zsh) as your login shell and launch xonsh from its RC file, or use    
  `wsl.exe -e /path/to/xonsh` to bypass the default shell entirely.

**Comment by eugenesvk** [comment]
> it stops reading my `~/.config/xonsh/rc.xsh` file on manual startup

So how is this your solution?

> **Solution:** move your configuration to `~/.config/xonsh/rc.xsh`.

Re. potential breakages: that's fine for the the WSL

**Comment by anki-code** [comment]
To solve the issue we need more tests and may be code patch here:

https://github.com/xonsh/xonsh/blob/643d630d0fb1519ab70ab216044b0962cd2d5fc7/xonsh/main.py#L341

Could you please add the code before this return:

```xsh
import json, datetime                                                                                                                                                                 
   
with open("/tmp/xonsh_env_debug.log", "a") as f:                                                                                                                                      
    f.write(f"\n=== {datetime.datetime.now()} ===\n")           
    for k, v in env.items():
        try:
            f.write(f"{k}={v!r}\n")
        except Exception as e:
            f.write(f"{k}=<error: {e}>\n")
```
And then try to run with your WSL case.

**Comment by eugenesvk** [comment]
Are you using LLMs to write all responses for you in these PRs/issues??? That would explain the many mistakes and self-contradictions like the one pointed out earlier

**Comment by anki-code** [comment]
I have 4 years background on improving xonsh code and deep diving into the core logic. Now I setted up the expensive model that presetted to make deep dive into the code and help me review and release things faster. I'm presonally review and test the responces of the model and revalidate them based on my 4 years experience.

The mistakes you mentiond (that related my memory about your PRs) are completely on me, hah :) The model reviewed your approch much more complete in [5654](https://github.com/xonsh/xonsh/pull/5654#issuecomment-4174902159). 

Please stop being angry with me and instead help me and offer solutions based on the current situation, not the past.

**Comment by eugenesvk** [comment]
> that related my memory about your PRs

No, that's not the issue I was referencing, but the previous comment https://github.com/xonsh/xonsh/issues/4495#issuecomment-4147669603 from this very thread which references your comment offering as a "solution" the very same one that is causing the original issue! Another one is I've also quoted about your "learning" about PowerShell, which are the opposite of each other.

> instead help me and offer solutions based on the current situation, not the past.

Since you haven't actually solved any of the issues discussed earlier, the solutions offered in the past are identical to the ones that would work in the present!

But also, for this specific issue, there is no solution yet, so it should just remain opened as an unresolved issue

**Comment by anki-code** [comment]
I'm ok to reopen with the link to [issue review](https://github.com/xonsh/xonsh/issues/4495#issuecomment-4145691281) and [next step](https://github.com/xonsh/xonsh/issues/4495#issuecomment-4147698131).
